Migration from 14.8.6 to 15.0.4 makes our libs unbuildable with "is not under 'rootDir'" errors

## Current Behavior

After migrating from v14.8.6 to v15.04, some libraries start showing errors regarding `File is not under 'rootDir'`. Using the v14.8.6, it used to work fine. 

The reproduction repo contains three libraries:
* `libs/frame/common`
* `libs/frame/core/errors`
* `libs/frame/core/valid`.

When trying to build the lib `libs/frame/core/valid`, it fails with the following error:

```bash
Compiling TypeScript files for project "frame-core-valid"...
libs/frame/common/src/index.ts:1:15 - error TS6059: File '/Users/leand/github.com/leandro-branco/nx-rootdir-issue/libs/frame/common/src/lib/grpc-status.ts' is not under 'rootDir' 'libs/frame/core/valid'. 'rootDir' is expected to contain all source files.

1 export * from './lib/grpc-status'
                ~~~~~~~~~~~~~~~~~~~
libs/frame/common/src/index.ts:2:15 - error TS6059: File '/Users/leand/github.com/leandro-branco/nx-rootdir-issue/libs/frame/common/src/lib/http-status.ts' is not under 'rootDir' 'libs/frame/core/valid'. 'rootDir' is expected to contain all source files.

2 export * from './lib/http-status'
```

Note that the `libs/frame/core/valid` doesn't import any code from `libs/frame/common`, only from `libs/frame/core/errors.` If we force the import of the common library, the build works fine.

I've already tried to remove cache `rm -rf ./node_modules/.cache/nx` and reset nx and to include `"analyzeSourceFiles": true` into nx.json.

Not sure if it's related to https://github.com/nrwl/nx/issues/12338#issuecomment-1284276013 

## Expected Behavior
To build with no errors

## Steps to Reproduce

Clone the repro code: https://github.com/leandro-branco/nx-rootdir-issue
Run the commands:
```bash
yarn install
nx run frame-core-valid:build
```

### Environment

```bash
 >  NX   Report complete - copy this into the issue template

   Node : 16.16.0
   OS   : darwin x64
   yarn : 1.22.19
   
   nx : 15.0.4
   @nrwl/angular : 15.0.4
   @nrwl/cypress : 15.0.4
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.0.4
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.0.4
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.0.4
   @nrwl/js : 15.0.4
   @nrwl/linter : 15.0.4
   @nrwl/nest : 15.0.4
   @nrwl/next : 15.0.4
   @nrwl/node : 15.0.4
   @nrwl/nx-cloud : 15.0.0
   @nrwl/nx-plugin : 15.0.4
   @nrwl/react : 15.0.4
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.0.4
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.0.4
   @nrwl/web : 15.0.4
   @nrwl/webpack : 15.0.4
   @nrwl/workspace : 15.0.4
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
         @storybook/angular: 6.5.12
```

--- a/packages/js/src/utils/check-dependencies.ts
+++ b/packages/js/src/utils/check-dependencies.ts
@@ -19,8 +19,7 @@ export function checkDependencies(
     context.root,
     context.projectName,
     context.targetName,
-    context.configurationName,
-    true
+    context.configurationName
   );
   const projectRoot = target.data.root;
 

--- a/packages/workspace/src/utilities/buildable-libs-utils.ts
+++ b/packages/workspace/src/utilities/buildable-libs-utils.ts
@@ -137,7 +137,8 @@ function collectDependencies(
         return;
 
       acc.push({ name: dependency.target, isTopLevel: areTopLevelDeps });
-      if (!shallow) {
+      const isInternalTarget = projGraph.nodes[dependency.target];
+      if (!shallow && isInternalTarget) {
         collectDependencies(dependency.target, projGraph, acc, shallow, false);
       }
     }

